packages/smart-wallet/src/offers.js

```diff
@@ -6,12 +6,19 @@
  * @typedef {number | string} OfferId
  */
 
+/**
+ * @typedef {{ method: string; args?: unknown[] } | { saveAs: string }} OfferResultStep
```

Comment:
Somewhat concerned about how easy it may be to drop results on the floor. I would prefer that the `saveAs` be part of the "operation" step, be mandatory, but possibly accept a `null` value to explicitly opt-out of saving the operation's result.

Similarly I think that args can be mandatory, just be an empty array if the method takes no args.

Finally method may need to accept an `undefined` value if the target is a function (instead of a method).